slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -477,6 +487,16 @@ void SLAMServiceImpl::RunSLAM() {
 
         map_builder.map_builder_->FinishTrajectory(trajectory_id);
         map_builder.map_builder_->pose_graph()->RunFinalOptimization();
```

Comment:
[q] Doesn't running final optimization when extracting pointcloud introduce a time lag? I thought thats one reason why we have decided on using a optimization flag

Reply:
This is for at the end of the dataset in offline mode. Not sure if we want to keep it here but was leaving it in anyways. Currently we would not be able to update our map while this optimization occurs, which has made testing fun. I think the trick is making GetMap() not rely on the `map_builder_mutex`, which we have an [investigation ticket](https://viam.atlassian.net/browse/DATA-668) for

Reply:
oh also this part isn't in the grpc GetMap() call, this is in `RunSLAM()`

Reply:
We discussed when and where to do optimization, either at the beginning with a potential long delay before being able to actually start or at the end and requiring users to close out of slam and wait before closing out the server. The result of the conversation as I remember was to do it at the beginning in the optimize flag was true as we can't control a user's behavior at the end of a run. 

This code might be an artifact from our initial implementation of slam when we were always doing optimization at the end. If you dont want to handle it in this PR, I'd recommend making a ticket

Reply:
yeah I think it makes sense to move/change this on the ticket for implementing the optimize flag. Is the start/end behavior written down somewhere?

Reply:
Discussed here: https://docs.google.com/document/d/1LdD_q20timS7nUF43Aqa1fJ__p8N1u8C-NVVT6C0V9I/edit#bookmark=id.r6yew9ly8e33

Reply:
I don't see harm in saving an optimized map at the end of a successful offline run.

Reply:
An example of this might be that I run through an offline dataset, and I want to see the optimized map - which looks different from before the optimization. Without optimizing the map at the end, there is no way for a user to view the optimized map.

Reply:
I agree some what, however it goes against the tech spec so we need to discuss if we want to include it.

Options:
1. Do it and document that this occurs in offline but not online
2. Don't do it and require a user to load the map up and set optimize_on_start flag to perform operation and then save (not sure exactly how they would play out with no new data, although we did discuss doing it at the time)
3. introduce a optimize_at_end flag for explicit saving, offline or online

Reply:
I think it should be option 1. I do also think that this should be put on another ticket rather than done in this PR because it isn't related to the task at hand.